web/src/pages/api/public/media/[mediaId].ts

```diff
@@ -0,0 +1,123 @@
+import { z } from "zod";
+
+import { env } from "@/src/env.mjs";
+import { getMediaStorageServiceClient } from "@/src/features/media/server/getMediaStorageClient";
+import {
+  GetMediaQuerySchema,
+  GetMediaResponseSchema,
+  PatchMediaBodySchema,
+} from "@/src/features/media/validation";
+import { createAuthedAPIRoute } from "@/src/features/public-api/server/createAuthedAPIRoute";
+import { withMiddlewares } from "@/src/features/public-api/server/withMiddlewares";
+import {
+  ForbiddenError,
+  InternalServerError,
+  LangfuseNotFoundError,
+} from "@langfuse/shared";
+import { Prisma, prisma } from "@langfuse/shared/src/db";
+import { recordIncrement, recordHistogram } from "@langfuse/shared/src/server";
+
+export default withMiddlewares({
+  GET: createAuthedAPIRoute({
+    name: "Get Media data",
+    querySchema: GetMediaQuerySchema,
+    responseSchema: GetMediaResponseSchema,
+    fn: async ({ query, auth }) => {
+      if (auth.scope.accessLevel !== "all") throw new ForbiddenError();
+
+      const { projectId } = auth.scope;
+      const { mediaId } = query;
+
+      const media = await prisma.media.findFirst({
+        where: {
+          projectId,
+          id: mediaId,
+        },
```

Comment:
Would it make sense to add an index for this query as well? Right now, it doesn't seem to be covered.

Reply:
The `id` column is unique and already indexed, I added the `projectId` here for access safety only, query should be highly performant without a composite index already :)